### DISCUSSION TITLE
laravel/react - Inertia.get reloading page

### DISCUSSION THREAD

**Discussion by ylandsman** [inertiajs/inertia#813] [Help (React)]
Please bear with me as I am new to both react and InertiaJs

I have a button that is calling an onclick function.  The onclick calls the following function

>  const editProduct=(e) => {

>         Inertia.get(route('admin.edit', e.currentTarget.value))
> 
>    }

When I click the button, the page refreshes, and I get a new page that has just this component.  What I want is for the instance of this component that is already loaded on my page to be the recipient of this request.  What am I missing?

I should note, that the component I want to load the data into is a material-ui modal dialog, that I want to have overlaying the page it is opened frm

Thanks in advance.

**Comment by ajnsn** [comment]
@ylandsman Take a look at the `preserveState` option, documentation [over here](https://inertiajs.com/manual-visits#component-state).

**Reply by ylandsman** [reply]
My call now looks like this:-

  ```
const editProduct=(e) => {
        console.log("edit product")
        Inertia.get(route('admin.edit', e.currentTarget.value), {}, {preserveState: true} )
    }
```

Before pressing the button, my component tree looks like this:-

![before](https://user-images.githubusercontent.com/2855518/125946559-05fac041-6228-40f4-b219-38759ec64364.png)

afterwards

![after](https://user-images.githubusercontent.com/2855518/125946810-61abd988-4cd4-4c4e-adf9-10c0d667cf34.png)

**Reply by ajnsn** [reply]
@ylandsman Ah sorry, my bad. I should have read correctly. You want to load a modal on an existing page. Thats quite challenging with Inertia but there's an [open PR](https://github.com/inertiajs/inertia/pull/642) which might make this feature available some day.

You might want to use a direct XHR request or try some of the workarounds, e.g. [here](https://github.com/inertiajs/inertia/discussions/249).

**Comment by ylandsman** [comment]
Strangely enough, for creating a record I have no problem.  When I do the post, I close the modal, and the new record shows up on the next render cycle.  Where I am struggling is on the retrieval of the record before editing.  Maybe a direct xhr request is the answer, although that sounds like it would make my app kind of hybrid.  Thanks for your help.

Yehuda Landsman
